### ISSUE TITLE
Ability to skip peds with World.GetClosestPed and family

### ISSUE THREAD

**Issue by justalemon** [scripthookvdotnet/scripthookvdotnet#1336]
World.GetClosestPed allows you to filter specific Model's, but not Ped's, which makes them unusable for finding peds around or near the player, as the player would be the first one to be found at that location.

Would it be possible to get another methods that allows you to filter by ped instead? Something like

```cs
public static Ped GetClosestPed(Vector3 position, float radius, params Ped[] peds);
public static Ped[] GetNearbyPeds(Ped ped, float radius, params Peds[] peds);
public static Ped[] GetNearbyPeds(Vector3 position, float radius, params Peds[] peds);
```

**Comment by kagikn** [comment]
Yeah, such additional methods or overloads that with an parameter array of exclude peds would be worth it. I was annoyed with how `World.GetClosest*` can't filter out the player ped or vehicle, too actually. I'm going to implement them and let's see how people respond to them…

I'm thinking it may be good to introduce builder classes for search queries for entity pools now, but I'm not implementing such builder classes right away. Needs to implement them with a lot of attention, but I want the console to be able to display a string in multiple lines first.

**Comment by kagikn** [comment]
I ran into an issue on method names, and We'll have to give new methods other names to avoid the overload ambiguity, such as `GetClosestPedNoExcludePeds`. We can't define overloads for existing `GetNearby*` or `GetClosest*` with the 3rd param with the `Ped[]` type so the compiler can determine an appropriate overload even if you pass null as the 3rd arg. Compiled scripts wouldn't get throw any exception for the ambiguity since methods with a params array parameter would be resolved as methods with `ParamArrayAttribute`, but still it'll be inconvenient.

**Comment by Tanjitsu** [comment]
Not efficient but I just use...

`World.GetClosestPed(Game.Player.Character.Position, 30f, Array.ConvertAll<Ped, Model>(peds, ped => new Model(ped.Model.Hash)));`
